### PR TITLE
Fix some functional test issues running locally.

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Fixes some functional tests that can fail intermittently.
Fixes path issues on local machine.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/api_validation_test.go [hunk 1]
```diff
@@ -24,8 +26,23 @@ const (
...
+
 	f := framework.NewFrameworkOrDie("api-validation-func-test")
 
+	dir, err := os.Getwd()
+	if err != nil {
+		Fail("Unable to get current directory")
+	}
+
+	// If we are in the 'tests' directory already, go up a level. This happens in the dev environment.
+	if strings.HasSuffix(dir, "tests") {
```
Comment (mhenriks) [inline tests/api_validation_test.go]: Is there any way we can address in bash somewhere?  Like in `run-tests.sh`?  Thinking for the case when we are running in ci, we can cd to the tests dir?  This doesn't seem like a sustainable pattern.

Comment (awels) [reply tests/api_validation_test.go]: I looked at that first, the problem is that in CI we call 
make functional-test-ci which generates a tests binary, and it is run from the root of the project. But when running in dev environment we call go test tests/... which apparently cds into the tests directory and then runs the tests. Not sure how we would get out of that from bash.

Comment (mhenriks) [reply tests/api_validation_test.go]: Yeah I ran into this the other day, and here's how I got around it:  https://gist.github.com/mhenriks/d25916aadd369d1d30e362618ce3cde2

That got `make test-functional` working for me and should still be good for ci.  Maybe there's an even better way to do it?

Comment (awels) [reply tests/api_validation_test.go]: Actually that looks good to me, and avoids me having to do silly stuff like I am doing there.

**Comment by awels** [comment]
ci test please

**Comment by mhenriks** [comment]
ci test please

**Pull request merged**